### ISSUE TITLE
Error in UseMethod("compile") :    no applicable method for 'compile' applied to an object of class "NULL"

### ISSUE THREAD

**Issue by Angel-Wei** [rstudio/keras#770]
I built a model using keras 2.2.4 but I got compile error like this. I tried to downgrade the R version from 3.5 to 3.4 and tried devtools::install_github("rstudio/keras") and "reticulate". Really can`t figure out what`s the problem. The script used to run well on another computer.
   
<img width="873" alt="Screen Shot 2019-05-09 at 3 05 48 PM" src="https://user-images.githubusercontent.com/39067474/57479681-f8a2d980-726b-11e9-9ce8-eacc004cd511.png">

**Comment by dfalbel** [comment]
Can you share your model building code too? It seems to me that there was a problem when building it.

**Comment by Angel-Wei** [comment]
Yes. Here`s the code to build the model
<img width="1021" alt="Screen Shot 2019-05-09 at 3 54 27 PM" src="https://user-images.githubusercontent.com/39067474/57482513-d06aa900-7272-11e9-80b3-c87f4a7cd4b3.png">

**Comment by dfalbel** [comment]
Can you share the code as text? So I can copy paste and test it in my setup. I can't see in your image if there was an error too.

**Comment by Angel-Wei** [comment]
Thanks for the help.
model <- keras_model_sequential() %>%
  layer_conv_1d(filters = 100, kernel_size = 8, strides = 1,padding = "valid",input_shape = c(101,4), activation = "relu") %>%
  layer_dropout(0)%>%
  
  layer_conv_1d(filters = 100, kernel_size = 6, strides = 1,padding = "valid", activation = "relu") %>%
  layer_dropout(0.7)%>%
  # 
  # layer_conv_1d(filters = 70, kernel_size = 8, strides = 1,padding = "valid", activation = "relu") %>%
  # layer_dropout(0.7)%>%
  
  layer_max_pooling_1d(pool_size = 4, strides = 4)%>%
  layer_dropout(0.6)%>%
  
  layer_lstm(units = 15, return_sequences = TRUE, use_bias = TRUE, activation = "relu") %>%
  layer_dropout(0.6) %>%  
  
  layer_flatten() %>%
  layer_dense(units = 2, activation = "sigmoid") %>%
  
  ## Geta summary of the model
  summary(model)

model %>% compile(
  loss = "binary_crossentropy",
  optimizer = optimizer_adamax(),
  metrics = c("accuracy")
)

checkpointer = callback_model_checkpoint(filepath = "saved_model/R_3.hdf5", monitor = 'val_acc', verbose = 1, save_best_only = TRUE);

###
history <- model %>% fit(
  x_train_array_new,
  y_train,
  epochs = 70, 
  batch_size = 64,
  validation_split = 0.2,
  callbacks = checkpointer
)
#model = load_model_hdf5(filepath = "FMRP_R_1.hdf5");

model %>% evaluate(x_test_array_new,y_test, verbose = 0)

**Comment by Angel-Wei** [comment]
model <- keras_model_sequential() %>%
  layer_conv_1d(filters = 100, kernel_size = 8, strides = 1,padding = "valid",input_shape = c(101,4), activation = "relu") %>%
  layer_dropout(0)%>%
  
  layer_conv_1d(filters = 100, kernel_size = 6, strides = 1,padding = "valid", activation = "relu") %>%
  layer_dropout(0.7)%>%
  
  layer_max_pooling_1d(pool_size = 4, strides = 4)%>%
  layer_dropout(0.6)%>%
  
  layer_lstm(units = 15, return_sequences = TRUE, use_bias = TRUE, activation = "relu") %>%
  layer_dropout(0.6) %>%  
  
  layer_flatten() %>%
  layer_dense(units = 2, activation = "sigmoid") %>%
  
  summary(model)

model %>% compile(
  loss = "binary_crossentropy",
  optimizer = optimizer_adamax(),
  metrics = c("accuracy")
)

checkpointer = callback_model_checkpoint(filepath = "saved_model/R_3.hdf5", monitor = 'val_acc', verbose = 1, save_best_only = TRUE);

history <- model %>% fit(
  x_train_array_new,
  y_train,
  epochs = 70, 
  batch_size = 64,
  validation_split = 0.2,
  callbacks = checkpointer
)

model %>% evaluate(x_test_array_new,y_test, verbose = 0)

**Comment by dfalbel** [comment]
It seems that you have an extra pipe (`%>%`) in the end of your model build block.

**Comment by Angel-Wei** [comment]
It works fine now! The code works fine with an extra pipe(%>%) on another computer. It`s weird I have to delete it to run on my PC. New to R. Thanks for the help! Appreciate it!
